app/searchers/searcher.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+# Abstract class for searching a relation via query params.
```

Comment:
More comments here about diff between this and Search. Search knows less, just how to take query string and qualifiers and convert into SQL>